Update wording to reflect Infrastructure Packages

brikis98 opened the pull request:
1. Use the term Infrastructure Package everywhere
1. Link to the module-xxx-public repos
1. Add icons for packages

@josh-padnick See if you like the new wording for each package. I’ll attach a screenshot shortly so you can see it in context with the new icons. If it looks good, we should update the wording in each repo to match. Likewise, we should update the repos to use the term “Infrastructure Package”… The one down side is we named them module-XXX instead of package-XXX… And renaming would not be backwards compatible :(

brikis98 commented:
Here is a screenshot:

![screenshot](https://cloud.githubusercontent.com/assets/711908/17081135/8b09a54a-511a-11e6-83dd-8b65914a2ac7.png)

josh-padnick commented:
First, thanks for taking the lead on this! 

I ran this branch on my localdev (nothing more than `docker-compose up` required!) to browse the new site, and before reviewing any of the specific file changes here's my feedback:

1. It's pretty cool to see a list of *real* packages versus the pseuo-packages that we had before. 

1. I'm very much in favor of publishing our package docs, but I'm on the fence now about whether linking from our website really makes sense. The primary issue I have is that we give you many different link opportunities to go into a content black hole.
  
   I know we're not ecommerce where buyers are fickle and choose not to buy their kitchen knives b/c something shiny and new popped up, but I still worry about not having a "linear" path for our website visitors that leads them to our Contact Us.

  As an alternative, we could consider (a) publishing our repos but not linking them, (b) using a single link on a separate page that lists all the links at once and talks about the context in which we published all the docs, or (c) option b but on a page that's not linked.

  Option c might be useful if we want to tell clients to go to gruntwork.io/packages to browse them.

1. Personally, I'm not a fan of the package icons. What I liked about the previous ones was their consistent look & style, which made them feel unified as an offering (albeit a little generic).

   I find the different colors jarring, and the use of some of the logos (e.g. Jenkins and MySQL) misleading since those packages do much more than just those tech's.

   I do like Grunty for consulting! And I also like having a central centered icon versus the "to the left" smaller icon that we had before.

   I wonder if visually splitting each of the packages as its own self-contained border (versus a set of "connected" rectangles, would also highlight these as individual packages?

  As for the icons, I vote for a standardized vector style for now. I actually tried to look around online for a good example of a well-illustrated common set of packages, and I really couldn't find anything.

I'll keep reviewing the site for feedback, then will get into wording shortly.

brikis98 commented:
> The primary issue I have is that we give you many different link opportunities to go into a content black hole.

I have no reason to suspect that people that click on our links to learn more about our packages would then forget to contact us if they were interested. As you said, we're not an e-commerce site, and reaching out to us isn't an impulse purchase that is going to be derailed by minor distractions. In fact, the docs are part of the marketing materials and, hopefully, strengthen our sales pitch. If no one can find them, they do very little for us.

> I actually tried to look around online for a good example of a well-illustrated common set of packages, and I really couldn't find anything.

That's exactly the problem. I wanted a set of standardized, high-res, lovely icons that fit our use cases, but couldn't find any. So these ones are the best I could do in a reasonable time frame.

Not having any icons at all felt way too text heavy. People tend to scan webpages and icons communicate a lot of information very quickly and grab your attention. Paragraphs of text by themselves would probably be skipped, but the icons let me get the general idea "at a glance".

Of course, the real answer to this question would be to run an A/B test and see what performs better, but we don't have nearly enough traffic for that.

josh-padnick commented:
Haven't read your response yet, but just throwing more thoughts out:

1. The only reason a user would engage in the detailed docs would be if they're so interested in our approach that they want to see "what does this actually look like?" So maybe we should dangle the public docs as a carrot we use to get someone signed up on our mailing list.

   That is, we could say "Want to see the package docs?" And then they click to a from, fill out some very limited info (name, email, why they're interested, subscribe to newsletter?).

1. In general, one of my concerns is that there's a *lot* of information density here. I wonder if maybe it's too much? 

  The goal of the site isn't to close deals for us (at least for now), it's to get them interested enough to sign up for our newsletter or better yet contact us. Do these additions help us optimize that?

  My concern is that maybe there's too much detail here, more than is necessary to get a user over the hump of "maybe I should contact these guys?"

  For example, I feel like https://www.intercom.io does a decent job of funneling you to contact them. Their product didn't work for us, but I felt emotionally excited by it before we were ready to try it. I liked how they listed a bunch of features in a really simple, easy to understand way without me having to read much (just being honest).

  And then of course the next step is to sign up for the trial and we what we think.

A lot of thinking out loud here. Just brainstorming on this. Let me read your other comments now.

josh-padnick commented:
> I have no reason to suspect that people that click on our links to learn more about our packages would then forget to contact us if they were interested. 

I wonder if we're coming from different default positions. I think my default position is "least information necessary to get them to contact us." Maybe yours is "Share all possible relevant info?"

Maybe that's a good way to discuss this. Philosophically, on what basis should we include/exclude content from our site?

josh-padnick commented:
> That's exactly the problem. I wanted a set of standardized, high-res, lovely icons that fit our use cases, but couldn't find any. So these ones are the best I could do in a reasonable time frame.

Actually, I thought our previous vector icons were pretty good. I suppose they're "lo-res" in that they're vector, but they're simple and distinctive.

Also, looking around at almost any product/service site I can think of, almost everyone uses low-res vector-ish icons to punctuate their features. For example, at https://www.docker.com/enterprise#/docker-solution, the icons for "Operating Systems", "CI/CD", etc. appeal more to me for package icons.

Also, I think stylistic consistency is important. The current icons don't really seem tied together at all, whereas a set of icons from the same vector library tie together automatically nicely.

So, I vote for using stylized, vector icons in place of the current ones. We can discuss more real-time today if you want.

josh-padnick commented:
Also, I'm going to hold off on reviewing the wording of the packages until we agree philosophically on some of the content questions I raised earlier.

josh-padnick commented:
After thinking about it, these website updates open up a lot of broad and interesting questions:
- Under what conditions is a prospect likely to reach out to us?
- What's the essential info we need to communicate to them?
- What support info if any should we share?

I think these questions would be better-suited for real-time discussion. Perhaps later today?

brikis98 commented:
> I wonder if we're coming from different default positions. I think my default position is "least information necessary to get them to contact us." Maybe yours is "Share all possible relevant info?"

My goal is to get people interested enough to contact us. Personally, I would NEVER reach out to a random consulting-ish company, unless I had a lot of info about why I should trust them, what they do, etc. I simply cannot imagine seeing a site that says "we are DevOps consultants, we're wicked good, you should totally contact us" and actually reaching out. Likewise, when looking at any new product or service, like MailChimp or Intercom, I dig into their features, docs, team, etc before signing up. I need a lot of info before I just hand over my contact details.

>  Philosophically, on what basis should we include/exclude content from our site?

Again, the ideal answer is to A/B test and find out. Practically speaking, that's hard to do right now, so all I can do is guess based on what I look for when signing up for a product or service. And that's usually a combination of:

1. What do you do? I want to know as concretely as possible instead of vague sales pitches.
1. How do you do it? Show me how your system works. There is no magic, so anyone who sells themselves as a cure-all is immediately suspect.
1. Who are you? In other words, why should I trust you? This is a combination of who else trusts you and what is your background.

> In general, one of my concerns is that there's a lot of information density here. I wonder if maybe it's too much?

I doubt it, but a possible solution is to show a "teaser" of the information--the first few Packages, the first few clients--and then a "show all" button that uses JavaScript to reveal the rest to anyone interested. We can add tracking on that button to see how many visitors click it.

brikis98 commented:
> For example, I feel like https://www.intercom.io does a decent job of funneling you to contact them. 

Signing up for a ~$20/month SaaS product with a free trial is not directly comparable to buying thousands of dollars of software and hiring consultants, but the amount of info they have is the same or more. They have a products menu, how it works, features, videos, etc. And I looked at quite a bit of it and found it a bit too vague for my tastes, actually. This was confirmed when I started using the product and found it didn't do all that much.

brikis98 commented:
> Also, I think stylistic consistency is important. The current icons don't really seem tied together at all, whereas a set of icons from the same vector library tie together automatically nicely.

So, there are actually two questions here:

1. Should we use logos from some of the products we work with?
1. Should all the icons look the same stylistically? 

The reason I moved from our vector icons to company logos was I think that those logos have a powerful impact. People recognize the Docker, AWS, MySQL, Jenkins, etc logos, especially if they are looking for help with those exact technologies. Again, it lets you skip a bunch of text and recognize what the company is doing in an instant. With the vector icons, you had to read a whole paragraph before you knew what the "rocket icon" or "globe icon" meant in this case.

For example, note how Packer has the logos of companies they work with: https://www.packer.io/
Same with dokku: http://dokku.viewdocs.io/dokku/
Same with StackExpress: http://stackexpress.com/

As for the stylistic aspect, I agree that it doesn't look beautiful currently. Making all the icons a single palette (like you did for the company logos) or putting them on a white background or various other changes could fix that. However, my goal was to get this done fast, and each of these design battles takes a non-trivial amount of time. I've already burned up several days working on the blog post, site changes, newsletter, documentation generator, license files, etc etc and it feels like prettyfing the logos is something we could improve incrementally...

brikis98 commented:
OK, I've switched back to the vector icons for the package list and added a list of technologies that have the company logos:

![collapsed](https://cloud.githubusercontent.com/assets/711908/17127135/3f5668fa-52d1-11e6-9f57-db722edecdc0.png)

Each of the "cells" in the list of packages turns white when you hover over it. When you click it, it expands to show the text and documentation link:

![expanded](https://cloud.githubusercontent.com/assets/711908/17127152/5631bdd6-52d1-11e6-8dc8-b50b90336217.png)

Unless you have objections, I'll merge all this stuff tomorrow and release it along with the blog post.

josh-padnick commented:
These updates look awesome! It's a really nice improvement.

Some minor feedback to improve/refine even further:

1. When you expand "Network Topology" or any of the other packages and hover your mouse over it, I really like the white hover effect, but it does make the text a little harder to read. I wonder if `rgb(220,220,220)` would make it look better? Looked nice in my local web inspector.

TBD

brikis98 commented:
@josh-padnick The color change doesn't seem to help much. What do you think of the restyle in #4?

josh-padnick commented on _data/packages.yml:
```diff
@@ -1,42 +1,73 @@
+    Set up best security best practices for managing secrets, credentials, and servers. This Package uses AWS Key
+    Management Service (KMS) to encrypt and decypt secrets so you can securely store them version control, configures
+    servers to accept SSH connections from an authorized list of IAM users that updates automatically, and provides
+    server hardening with automated security updates and tools like fail2ban.
```
s/Set up best security best practices/Set up security best practices/

josh-padnick replied:
so you can securely store them *in* version control

brikis98 replied:
Good catches. Fixed: https://github.com/gruntwork-io/gruntwork-io.github.io/commit/3f6bfe327982fb9733f0d6d7c86dd2c0dbe9d46c

josh-padnick commented on _data/packages.yml:
```diff
@@ -1,42 +1,73 @@
+    every commit, this Package build your code, runs your tests, and packages your app into a deployable, versioned
+    artifact (e.g. Docker image or Amazon Machine Image). You can kick off an automated deployment by pushing to a
+    specific branch, creating a new tag, manually running a CI job, or even after every single commit (continuous
+    deployment).
```
Replace first sentence with:

Set up an automated pipeline to build and deploy code using a self-hosted CI server (e.g. Jenkins) or cloud-based CI service (e.g. CircleCI, or Travis).

josh-padnick replied:
After every commit, this Package *builds* your code

brikis98 replied:
Updated and fixed: https://github.com/gruntwork-io/gruntwork-io.github.io/commit/3f6bfe327982fb9733f0d6d7c86dd2c0dbe9d46c

josh-padnick commented on _data/packages.yml:
```diff
@@ -1,42 +1,73 @@
+  description: |
+    Run MySQL, Postgres, MariaDB, or Aurora on Amazon’s Relational Database Service (RDS). This Packages creates
+    the database, sets up replicas, configures automatic failover to a standby in a separate Availability Zone in case
+    of problems, and configures automatic backup.
```
or *Amazon* Aurora

brikis98 replied:
Added: https://github.com/gruntwork-io/gruntwork-io.github.io/commit/3f6bfe327982fb9733f0d6d7c86dd2c0dbe9d46c
